tests/std/tests/P1135R6_atomic_wait_vista/test.cpp

```diff
@@ -4,6 +4,11 @@
 #include "test_atomic_wait.hpp"
 
 int main() {
-    assert(__std_atomic_set_api_level(__std_atomic_api_level::__has_srwlock) == __std_atomic_api_level::__has_srwlock);
+    auto level = __std_atomic_set_api_level(__std_atomic_api_level::__has_srwlock);
+#if _ATOMIC_WAIT_ON_ADDRESS_STATICALLY_AVAILABLE
+    assert(level == __std_atomic_api_level::__has_wait_on_address);
+#else // ^^^ wait on address statically available ^^^ / vvv wait on address NOT statically available vvv
+    assert(level == __std_atomic_api_level::__has_srwlock);
+#endif // ^^^ wait on address NOT statically available ^^^

```

Comment:
1. If `__std_atomic_set_api_level` breaks this test will now continue to pass, tricking us into thinking we have coverage of the fallback when we don't.
2. This test is redundant with `P1136_atomic_wait` when `assert(level == __std_atomic_api_level::__has_wait_on_address);`.

I think I'd rather see the body of main wrapped in something like `#if !(defined(_M_ARM) || defined(_M_ARM64) || defined(_M_ARM64EC))`.

Reply:
Maybe check for x86 and x64 then instead? Future CPUs will have WaitOnAddress statically available too.

Reply:
> Maybe check for x86 and x64 then instead? Future CPUs will have WaitOnAddress statically available too.

Yes, that does make more sense.